src/sys/pthread.rs

```diff
@@ -34,10 +41,285 @@ pub fn pthread_kill<T>(thread: Pthread, signal: T) -> Result<()>
+        }
+    }
+    /// Wraps [`libc::pthread_mutexattr_getrobust`].
+    pub fn get_robust(&self) -> Result<bool> {
+        let init = unsafe {
+            let mut uninit = std::mem::MaybeUninit::uninit();
+            Errno::result(libc::pthread_mutexattr_getrobust(&self.0,uninit.as_mut_ptr()))?;
+            uninit.assume_init()
+        };
+        Ok(init == libc::PTHREAD_MUTEX_ROBUST)
+    }
+    /// Wraps [`libc::pthread_mutexattr_setrobust`].
+    pub fn set_robust(&mut self, robust: bool) -> Result<()> {
+        let robust = if robust { libc::PTHREAD_MUTEX_ROBUST} else { libc::PTHREAD_MUTEX_STALLED };
+        unsafe {
+            Errno::result(libc::pthread_mutexattr_setrobust(&mut self.0,robust)).map(drop)
+        }
+    }
+    /// Wraps [`libc::pthread_mutexattr_getprotocol`].
+    pub fn get_protocol(&self) -> Result<Protocol> {
+        let init = unsafe {
+            let mut uninit = std::mem::MaybeUninit::uninit();
+            Errno::result(libc::pthread_mutexattr_getprotocol(&self.0,uninit.as_mut_ptr()))?;
+            uninit.assume_init()
+        };
+        Ok(Protocol::from(init))
+    }
+    /// Wraps [`libc::pthread_mutexattr_setprotocol`].
+    pub fn set_protocol(&mut self, protocol: Protocol) -> Result<()> {
+        unsafe {
+            Errno::result(libc::pthread_mutexattr_setprotocol(&mut self.0,protocol as i32)).map(drop)
+        }
+    }
+}
+
+#[cfg(target_os = "linux")]
+impl Default for MutexAttr {
+    fn default() -> Self {
+        let mutex_attr = Self::new().unwrap();
+        debug_assert_eq!(mutex_attr.get_shared(),Ok(true));
+        mutex_attr
+    }
+}
+#[cfg(target_os = "linux")]
+impl std::ops::Drop for MutexAttr {
+    /// Wraps [`libc::pthread_mutexattr_destroy`].
+    fn drop(&mut self) {
+        unsafe {
+            Errno::result(libc::pthread_mutexattr_destroy(&mut self.0)).unwrap();
+        }
+    }
+}
+
+/// Mutex.
+/// ```
+/// # use std::{
+/// #   sync::Arc,
+/// #   time::{Instant, Duration},
+/// #   thread::{sleep, spawn},
+/// #   mem::size_of,
+/// #   num::NonZeroUsize,
+/// #   os::unix::io::OwnedFd
+/// # };
+/// # use nix::{
+/// #   sys::{pthread::{Mutex, MutexAttr}, mman::{mmap, MapFlags, ProtFlags}},
+/// #   unistd::{fork,ForkResult},
+/// # };
+/// const TIMEOUT: Duration = Duration::from_millis(500);
+/// const DELTA: Duration = Duration::from_millis(100);
+/// # fn main() -> nix::Result<()> {
+/// let mutex = Mutex::default();
+/// 
+/// // The mutex is initialized unlocked, so an attempt to unlock it should
+/// // return immediately.
+/// assert_eq!(mutex.unlock(), Ok(()));
+/// // The mutex is unlocked, so `try_lock` will lock.
+/// assert_eq!(mutex.try_lock(), Ok(true));
+/// // Unlock the mutex.
+/// assert_eq!(mutex.unlock(), Ok(()));
+/// // The mutex is unlocked, so `lock` will lock and exit immediately.
+/// assert_eq!(mutex.lock(), Ok(()));
+/// // Unlock the mutex.
+/// assert_eq!(mutex.unlock(), Ok(()));
+/// 
+/// // Test across threads
```

Comment:
These different examples should be in separate doc tests, not all combined together.  Or, in unit tests, if they aren't particularly useful as documentation.

Reply:
Fixed.